pandas/tests/frame/indexing/test_setitem.py

```diff
@@ -339,3 +339,17 @@ def test_setitem_boolean_mask(self, mask_type, float_frame):
         expected = df.copy()
         expected.values[np.array(mask)] = np.nan
         tm.assert_frame_equal(result, expected)
+
+    @pytest.mark.parametrize("func", [list, np.array, Series])
+    @pytest.mark.parametrize("value", [[], [False]])
+    def test_setitem_loc_only_false_indexer_dtype_changed(self, func, value):
+        # GH#37550
+        # Dtype is only changed when value to set is a Series and indexer is
+        # empty/bool all False
+        df = DataFrame({"a": ["a"], "b": [1], "c": [1]})
+        df.loc[func(value), ["b"]] = 10 - df["c"]
+        expected = DataFrame({"a": ["a"], "b": [1], "c": [1]})
+        tm.assert_frame_equal(df, expected)
+
+        df.loc[[False], ["b"]] = 9
```

Comment:
should `[False]` be `func(value)` as above?  if not, this might belong as a separate test